src/shared/util/logic.ts

```diff
@@ -276,14 +289,24 @@ const isLogicUnitSatisfied = (
 }
 
 const getCurrentValue = (
-  field: LogicFieldSchemaOrResponse,
-): string | null | undefined | string[] => {
+  field: FieldSchemaOrResponse,
+):
+  | string
+  | string[]
+  | boolean[]
+  | ITableRow[]
```

Comment:
Done! I’ve added a function to check that the fields in `isConditionFulfilled` are logic fields. This is called before calling `getCurrentValue` so that we can narrow the types there.